akka-actor/src/main/scala/akka/actor/dungeon/DeathWatch.scala

```diff
@@ -11,7 +11,7 @@ import akka.event.AddressTerminatedTopic
 
 private[akka] trait DeathWatch { this: ActorCell ⇒
 
-  private var watching: Set[ActorRef] = ActorCell.emptyActorRefSet
+  private var watching: Map[ActorRef, Option[Any]] = Map.empty
```

Comment:
Should this perhaps be wrapped in a data structure that guarantees it isn't touched in the wrong way rather than by just remembering to use the mutator methods below?